### ISSUE TITLE
: There appear to be 1 leaked semaphores to clean up at shutdown\

### ISSUE THREAD

**Issue by rabeehk** [pytorch/xla#2647]
Hi
I would like to save some checkpoint and getting this warning which results in not saving the properly at the of a program,
I tried with adding xm.rendezvous('done') as below, still not working, could you tell me how I can solve this warning? thanks 

if trainer.is_world_process_zero():
        if training_args.gcs_bucket is not None:
            os.system("gsutil -m cp -r {} {}".format(training_args.output_dir,
                      os.path.join(training_args.gcs_bucket, training_args.output_dir)))
xm.rendezvous('done')

{
  "textPayload": "/root/anaconda3/envs/pytorch/lib/python3.6/multiprocessing/semaphore_tracker.py:143: UserWarning: semaphore_tracker: There appear to be 1 leaked semaphores to clean up at shutdown\n",
  "insertId": "hdonvak9jsjlaob7l",
  "resource": {
    "type": "k8s_container",
    "labels": {
      "namespace_name": "rabeeh",
      "cluster_name": "xcloud-v3-donut-europe-west4-a",
      "location": "europe-west4-a",
      "pod_name": "20201124.proj32.c8701.0-xxk8b",
      "project_id": "try-ideas-for-rmi",
      "container_name": "proj32"
    }
  },
  "timestamp": "2020-11-24T21:28:01.466105414Z",
  "severity": "ERROR",
  "labels": {
    "k8s-pod/jobowner": "rabeeh",
    "k8s-pod/app": "xcloud",
    "k8s-pod/serviceName": "xc-20201124-proj32-c8701-0",
    "k8s-pod/controller-uid": "35f74b1c-51bc-400f-a54b-f56474f59ee2",
    "k8s-pod/job-name": "20201124.proj32.c8701.0"
  },
  "logName": "projects/try-ideas-for-rmi/logs/stderr",
  "receiveTimestamp": "2020-11-24T21:28:05.328675396Z"
}

**Comment by zcain117** [comment]
The semaphore warnings shouldn't be fatal AFAIK. Have you tried `xm.save` to save things during training?

https://pytorch.org/xla/release/1.7/index.html#torch_xla.core.xla_model.save

**Comment by taylanbil** [comment]
@rabeehk is this good to close?

**Comment by rabeehkarimimahabadi** [comment]
yes, thanks.
